src/core_plugins/kibana/public/home/components/home_app.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Home } from './home';
+import { FeatureDirectory } from './feature_directory';
+import {
+  HashRouter as Router,
+  Switch,
+  Route
+} from 'react-router-dom';
+
+export function HomeApp({ addBasePath, directories, directoryCategories }) {
+  return (
+    <Router>
+      <Switch>
+        <Route
+          path="/home/feature_directory"
+        >
+          <FeatureDirectory
+            addBasePath={addBasePath}
+            directories={directories}
+            directoryCategories={directoryCategories}
+          />
+        </Route>
+        <Route
+          path="/"
+        >
+          <Home
+            addBasePath={addBasePath}
+            directories={directories}
+            directoryCategories={directoryCategories}
+          />
+        </Route>
+      </Switch>
+    </Router>
+  );
+}
+
+HomeApp.propTypes = {
+  addBasePath: PropTypes.func.isRequired,
+  directories: PropTypes.object.isRequired,
+  directoryCategories: PropTypes.object.isRequired,
```

Comment:
I like how this is dynamic so you can easily add categories but I don't think having it as an object will play nicely when typescript comes along (eventually). If you make it an array of strings it would be less flexible in terms of what could get passed and clearer to the reader what the requirements are.  Thoughts?

Reply:
Not sure an array will work. I need to have a named variable to access certain constants. For example, I need to get the directoryCategories.DATA to filter the featureCatalogue by category 'data'

Reply:
I was thinking something like this:
```
      $scope.directoryCategories = [
        FeatureCatalogueCategory.ADMIN,
        FeatureCatalogueCategory.DATA,
        FeatureCatalogueCategory.OTHER,
      ];
```

But now that I look again, the dynamic aspect of doing it this way wouldn't even work because in FeatureDirectory you check for these three values specifically.  I was originally thinking, well if a plugin wants to add a new category, so they register an app with category 'foo' and then you'd pass in 'foo' here, it would automatically create a new tab filled with those categories.  Though if you really wanted it dynamic you'd have to have a category name, and id, for display purposes.  And now I think I'm just overcomplicating it with these thoughts.

To make it simpler, why pass this as a prop at all, and not just access FeatureCatalogueCategory enum directly?

Reply:
no reason at all. I will update to just bring these in with an include